src/QMCDrivers/tests/test_dmc_omp.cpp

```diff
@@ -0,0 +1,146 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2018 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by:  Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "catch.hpp"
+
+
+#include "Utilities/RandomGenerator.h"
+#include "OhmmsData/Libxml2Doc.h"
+#include "OhmmsPETE/OhmmsMatrix.h"
+#include "Lattice/ParticleBConds.h"
+#include "Particle/ParticleSet.h"
+#include "Particle/DistanceTableData.h"
+#include "Particle/DistanceTable.h"
+#include "Particle/SymmetricDistanceTableData.h"
+#include "Particle/MCWalkerConfiguration.h"
+#include "QMCApp/ParticleSetPool.h"
+#include "QMCApp/HamiltonianPool.h"
+#include "QMCApp/WaveFunctionPool.h"
+#include "QMCWaveFunctions/OrbitalBase.h"
+#include "QMCWaveFunctions/TrialWaveFunction.h"
+#include "QMCWaveFunctions/ConstantOrbital.h"
+#include "QMCHamiltonians/BareKineticEnergy.h"
+#include "Estimators/EstimatorManagerBase.h"
+#include "Estimators/TraceManager.h"
+#include "QMCDrivers/DMC/DMCOMP.h"
+
+
+#include <stdio.h>
+#include <string>
+
+
+using std::string;
+
+namespace qmcplusplus
+{
+
+TEST_CASE("DMCOMP", "[drivers][dmc]")
+{
+
+  Communicate *c;
+  OHMMS::Controller->initialize(0, NULL);
+  c = OHMMS::Controller;
+
+  ParticleSet ions;
+  MCWalkerConfiguration elec;
+
+  ions.setName("ion");
+  ions.create(1);
+  ions.R[0][0] = 0.0;
+  ions.R[0][1] = 0.0;
+  ions.R[0][2] = 0.0;
+
+  elec.setName("elec");
+  elec.setBoundBox(false);
+  std::vector<int> agroup(1);
+  agroup[0] = 2;
+  elec.create(agroup);
+  elec.R[0][0] = 1.0;
+  elec.R[0][1] = 0.0;
+  elec.R[0][2] = 0.0;
+  elec.R[1][0] = 0.0;
+  elec.R[1][1] = 0.0;
+  elec.R[1][2] = 1.0;
+  elec.createWalkers(1);
+
+  SpeciesSet &tspecies =  elec.getSpeciesSet();
+  int upIdx = tspecies.addSpecies("u");
+  int downIdx = tspecies.addSpecies("d");
+  int chargeIdx = tspecies.addAttribute("charge");
+  int massIdx = tspecies.addAttribute("mass");
+  tspecies(chargeIdx, upIdx) = -1;
+  tspecies(chargeIdx, downIdx) = -1;
+  tspecies(massIdx, upIdx) = 1.0;
+  tspecies(massIdx, downIdx) = 1.0;
+
+  elec.addTable(ions,DT_AOS);
+  elec.update();
+
+  CloneManager::clear();
```

Comment:
I prefer to put this at the end if possible.

Reply:
With putting it at the end, I'm concerned that if one test exits early due to a failure, the 'clear' won't get called, and it could lead to a cascading failure in a different test.